### ISSUE TITLE
Questions about fine-tuning based on small dataset about  picking objects

### ISSUE THREAD

**Issue by Cyber-Alligater** [openvla/openvla#64]
Hey,

Thanks for your open-source code!! I am trying to fine-tune the vla model using my small dataset but have some questions:

_My small dataset for fine-tuning: 120 demons about pick up a yellow or a red object (fake fruits) .
Robot Arm: UR5_

**Q(1)** How to use **proprio** information of the robot? You mentioned in an other closed issue: “Adding proprio should be a small code change to the RLDSTransform”
Could you maybe describe it more specific about how to make the change?

(This is how I did, and maybe I misunderstood)
I changed  in line 46-48 in 
https://github.com/openvla/openvla/blob/f7e891580a2fbd429f5f1cc22a4ccdb3cdb71ad1/prismatic/vla/datasets/datasets.py#L48

from:

	 conversation = [
            {"from": "human", "value": f"What action should the robot take to {lang}?"},
            {"from": "gpt", "value": self.action_tokenizer(action)},
        ]

To:
         xxxx = proprio_data

	 conversation = [
            {"from": "human", "value": f"What action should the robot take to {lang}, given robot proprioception {xxxx} ?"},
            {"from": "gpt", "value": self.action_tokenizer(action)},
        ]

I tried with this, but then overfitting happens, action acc went to 100% and robot can do noting in inference tests.

**Q(2)** I guess you set **image_aug**=True when doing fine-tuing. 
In my case if I set image_aug=False, Action acc can reach over 95%. But if I set image_aug=True, Action acc will get stuck around 80% can almost not increase anymore. What can be the reason?

**Q(3)** Can lora_dropout help avoid overfitting, do you use in fine-tuning in your experiment?

**Q(4)** Since I also have **Depth** information, It can happen that robot always fail to grasp the object. Because from the _camera view,_ the object is located at middle of gripper and gripper should be able to grasp the object, and in fact, the object is somewhat behind the gripper.
Have you got similar issue during testing? Any ideas?

**Comment by kpertsch** [comment]
Hi, thanks for reaching out!

Re proprio: your implementation looks correct, just add it to the prompt! We have anecdotally observed that adding proprio may indeed hurt with small datasets, so maybe it's better to not add it -- do things work better without?

Re image aug: we do usually use image augmentation. i am surprised the model would struggle to fit the data with image aug. Is the robot rollout performance any different between the runs with and without augmentation?

Re lora dropout: we don't use that in our experiments -- we use the finetuning script as provided.

Re depth: this kind of 'early grasping' is a common failure mode. You can make sure that the lighting and camera angle between data collection and robot rollout tests is consistent (since the model may use shadows or distance to frame boundary to determine depth implicitly). You can also try changing the camera angle so it's less ambiguous?

Closing for now -- feel free to reopen if you have more quesitons!
